src/runtime/util.cpp

```diff
@@ -46,25 +46,25 @@ void parseSlice(BoxedSlice* slice, int size, i64* out_start, i64* out_stop, i64*
         }
     }
 
-    if (isSubclass(start->cls, int_cls)) {
-        istart = static_cast<BoxedInt*>(start)->n;
-        if (istart < 0)
-            istart = size + istart;
-    } else {
+    if (start->cls == none_cls || (start->cls != int_cls && !isSubclass(start->cls, int_cls))) {
```

Comment:
Do we need these changes?  it looks like it still has the same behavior but I hate touching the slice parsing code...